### PR TITLE
Handle Unrecognized Faces [GG-77]

### PR DESCRIPTION
**Pull request by kemptonk**
Implemented functionality that allows a person to authenticate themself by entering their employee ID and keycode.

### CODE REVIEW

File: FaceID/faceID.py [hunk 1]
```diff
@@ -37,6 +41,205 @@
...
 
+# Keypad layout
+KEYPAD = [
+    ["1", "2", "3", "A"],
+    ["4", "5", "6", "B"],
+    ["7", "8", "9", "C"],
+    ["*", "0", "#", "D"]
+]
+COL_PINS=[17, 15, 14, 4]
+ROW_PINS=[24, 22, 27, 18]
```
Comment (orndorffc) [inline FaceID/faceID.py]: COL_PINS = [17, 15, 14, 4]
ROW_PINS = [24, 22, 27, 18]

**Review by orndorffc — APPROVED** [review]
Ready to merge.

File: FaceID/I2C_LCD_driver.py [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+      self.lcd_device.write_cmd(data | LCD_BACKLIGHT)
+      self.lcd_strobe(data)
+
+   # write a command to lcd
+   def lcd_write(self, cmd, mode=0):
+      self.lcd_write_four_bits(mode | (cmd & 0xF0))
+      self.lcd_write_four_bits(mode | ((cmd << 4) & 0xF0))
+
+   # write a character to lcd (or character rom) 0x09: backlight | RS=DR<
+   # works!
```
Comment (lb-cake) [inline FaceID/I2C_LCD_driver.py:123]: I'd hope so! ;)

Comment (lb-cake) [reply FaceID/I2C_LCD_driver.py:123]: Nitpicking, but this could be removed

File: FaceID/faceID.py [hunk 3]
```diff
@@ -37,6 +41,205 @@
...
+
+# Whether the keypad should be accepting input
+acceptInput = False
+# Sequence of input keypresses
+code = ""
+# Variable used to update screen display
+mylcd = I2C_LCD_driver.lcd()
+
+# Retrieve authorization token for api and create header
+data = {"username":"admin", "password":"V@r3nTech#"}
```
Comment (lb-cake) [inline FaceID/faceID.py:66]: can this be retrieved from the settings.py file?

File: FaceID/faceID.py [hunk 4]
```diff
@@ -37,6 +41,205 @@
...
+        return -1
+
+
+# Retrieves and verifies the emp_ID
+def getEmpID():
+
+    global mylcd
+    global code
+    global url
+    global headers
```
Comment (lb-cake) [inline FaceID/faceID.py:173]: I'm not super stoked about having all these global variables, but for the sake of time, it works. Just for the future, avoid global variables as much as you can :)

**Review by lb-cake — APPROVED** [review]
Huge improvement on the code style. I love how your functions have gotten a lot cleaner.

I'll approve and merge due to time constraints, but just be mindful of the recommendations.

**Pull request merged**